connect/transforms/src/main/java/org/apache/kafka/connect/transforms/MaskField.java

```diff
@@ -119,6 +148,14 @@ else if (value instanceof Map)
         return maskedValue;
     }
 
+    private static Object maskWithCustomReplacement(Object value, String replacement) {
+        Function<String, ?> replacementMapper = REPLACEMENT_MAPPING_FUNC.get(value.getClass());
+        if (replacementMapper == null) {
+            throw new DataException("Cannot mask value of type: " + value.getClass() + " with custom replacement.");
+        }
+        return replacementMapper.apply(replacement);
```

Comment:
This call to the replacement mapping function may result in an exception, for example if the replacement value of `foo` is used on a field of type `DECIMAL`. It is proper to throw an exception, but would we not want to wrap that in a `DataException`? 

And speaking of failed replacements, it'd be good to have more such conversion failures in the `testReplacementTypeMismatch` unit test.